runtime/gateway.go

```diff
@@ -378,6 +379,14 @@ func (gateway *Gateway) handleHealthRequest(
 	req *ServerHTTPRequest,
 	res *ServerHTTPResponse,
 ) {
+	if gateway.isUnhealthy {
```

Comment:
consider writing a test, calling shutdown from one goroutine and another one calling health to check the expectation?

Reply:
I thought about it but I can't see a way to make the test deterministic.

Reply:
though it may be a pretty involved effort: there can be few ways, 1 way is - to mock HTTPServer (which is a dependency to a gateway) and have it return after you return from a health call with increasing the cancel timeout to a larger value like 1 min. this way you are guaranteed to run it reliably.

Reply:
as long as some aspect of timing is involved, I wouldn't say that is a guarantee.

Reply:
long cancel time is as good as a guarantee. however, if you are looking for a solution that is absolute, then you pass Shutdown context, which you can construct in your test without a timeout.